test/parallel/test-async-hooks-close-during-destroy.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+// Test that async ids that are added to the destroy queue while running a
+// `destroy` callback are handled correctly.
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+const async_hooks = require('async_hooks');
+
+const initCalls = new Set();
+let destroyTcpWrapCallCount = 0;
+let srv2;
+
+async_hooks.createHook({
+  init: common.mustCallAtLeast((id, provider, triggerId) => {
+    if (provider === 'TCPWRAP')
+      initCalls.add(id);
+  }, 2),
+  destroy: common.mustCallAtLeast((id) => {
```

Comment:
@refack Yeah, I kind of like it. :+1: You might want to re-review though, I’ve updated the test file quite a bit after Trevor’s comments in https://github.com/nodejs/node/pull/13286#discussion_r119496245